libraries/bsg_manycore_uart_responder.cpp

```diff
@@ -3,36 +3,82 @@
 #include <bsg_manycore_printing.h>
 #include <stdio.h>
 
-#define UART_EPA 0xEADC
+enum hb_mc_uart_epa_indx {
+  STDOUT_EPA_INDX, 
+  STDERR_EPA_INDX, 
+  BRANCH_TRACE_EPA_INDX,
```

Comment:
Can you put the branch trace responder code in a separate responder? 

I think STDERR and STDOUT are good in here. But the Branch Trace is performing a fundamentally different task

Reply:
Isn't it too much work to maintain multiple responders? I think having a generic mmio responder like this makes sense because, the only difference among handling mmios is what we print out to terminal. 
```
      switch (i) {
        case BRANCH_TRACE_EPA_INDX:
          fprintf(f, "BSG_BRANCH_TRACE x=%d y=%d target=%d\n", 
                  src_x, src_y, (int)data);
          break;

        default:
          fputc((int)data, f);
          break;
```
https://github.com/bespoke-silicon-group/bsg_f1/blob/8b121d4df53c69f16a89693a1bbf95bc5048f9fe/libraries/bsg_manycore_uart_responder.cpp#L77

Above switch is the only unique thing for each mmio epa.